packages/trader/src/Modules/CFD/Containers/cfd-dashboard.jsx

```diff
@@ -39,139 +39,124 @@ const LoadTab = ({ children, is_loading, loading_component, ...props }) => {
     return <Tabs {...props}>{children}</Tabs>;
 };
 
-class CFDDashboard extends React.Component {
-    state = {
-        is_demo_enabled: false,
-        is_real_enabled: false,
-        active_index: 0,
-        is_account_needed_modal_open: false,
-        is_demo_tab: true,
-        required_account: {},
-        is_notification_loaded: false,
-        password_manager: {
-            is_visible: false,
-            selected_login: '',
-            selected_account: '',
-            selected_account_type: '',
-            selected_account_group: '',
-            selected_server: '',
-        },
-    };
-
-    async componentDidMount() {
-        this.updateActiveIndex(this.getIndexToSet());
-        this.openResetPassword();
-        this.props.onMount();
-    }
-
-    componentWillUnmount() {
-        this.props.onUnmount();
-    }
-
-    componentDidUpdate() {
-        this.updateActiveIndex();
-        this.props.checkShouldOpenAccount();
-
-        if (this.props.is_logged_in) {
+const CFDDashboard = props => {
+    const [is_demo_enabled, setIsDemoEnabled] = React.useState(false);
+    const [is_real_enabled, setIsRealEnabled] = React.useState(false);
+    const [active_index, setActiveIndex] = React.useState(0);
+    const [is_demo_tab] = React.useState(true);
+    const [is_notification_loaded, setIsNotificationLoaded] = React.useState(false);
+    const [password_manager, setPasswordManager] = React.useState({
+        is_visible: false,
+        selected_login: '',
+        selected_account: '',
+        selected_account_type: '',
+        selected_account_group: '',
+        selected_server: '',
+    });
+
+    React.useEffect(() => {
+        updateActiveIndex(getIndexToSet());
+        openResetPassword();
+        props.onMount();
+        return () => {
+            props.onUnmount();
+        };
+    }, []);
+
+    React.useEffect(() => {
+        updateActiveIndex();
+        props.checkShouldOpenAccount();
+
+        if (props.is_logged_in) {
             ['demo', 'real'].forEach(account_type => {
                 const should_enable_tab =
-                    this.isSyntheticCardVisible(account_type) ||
-                    this.isFinancialCardVisible() ||
-                    this.isFinancialStpCardVisible();
+                    isSyntheticCardVisible(account_type) || isFinancialCardVisible() || isFinancialStpCardVisible();
 
-                const is_tab_enabled = this.state[`is_${account_type}_enabled`];
+                let is_tab_enabled = [`is_${account_type}_enabled`];
 
                 if (is_tab_enabled !== should_enable_tab) {
-                    this.setState({
-                        [`is_${account_type}_enabled`]: should_enable_tab,
-                    });
+                    is_tab_enabled = should_enable_tab;
```

Comment:
`is_tab_enabled` is not used anywhere. I think we should set the state here.
```
if (account_type === 'real') {
    setIsRealEnabled(is_tab_enabled);
} else if (account_type === 'demo') {
     setIsDemoEnabled(is_tab_enabled);
}
```

Reply:
I've set the states to should_enable_tab since our is_real_enabled and is_demo_enabled expect boolean value. Removed is_tab_enabled because as you mentioned it is a redundant line of code.